hudi-client/hudi-client-common/src/main/java/org/apache/hudi/io/HoodieMergeHandle.java

```diff
@@ -278,11 +279,17 @@ public void write(GenericRecord oldRecord) {
       HoodieRecord<T> hoodieRecord = new HoodieRecord<>(keyToNewRecords.get(key));
       try {
         Option<IndexedRecord> combinedAvroRecord =
-            hoodieRecord.getData().combineAndGetUpdateValue(oldRecord, useWriterSchema ? writerSchemaWithMetafields : writerSchema,
+            hoodieRecord.getData().combineAndGetUpdateValue(oldRecord,
+              useWriterSchema ? inputSchemaWithMetaFields : inputSchema,
```

Comment:
IIUC we do this so that the records keep getting written with the full table schema.

Reply:
yeah, the `HoodiePayload` use the `inputSchema` to parse the incoming record and compute the result record. The result record will have the full table schema.